llvm/utils/TableGen/DAGISelMatcherGen.cpp

```diff
@@ -573,8 +572,10 @@ bool MatcherGen::EmitMatcherCode(unsigned Variant) {
   // If the pattern has a predicate on it (e.g. only enabled when a subtarget
   // feature is around, do the check).
   std::string PredicateCheck = Pattern.getPredicateCheck();
-  if (!PredicateCheck.empty())
+  if (!PredicateCheck.empty()) {
+    CGP.increasePatternPredicateUsage(PredicateCheck);
```

Comment:
Seems unrelated?

Reply:
No, it's related. It's for increasing the usage of `PatternPredicate`.